src/utils/style-utils.js

```diff
@@ -1,139 +1,74 @@
-import isImmutableMap from './is-immutable-map';
-import diffStyles from './diff-styles';
 
 // TODO - remove in the next major release
 // Mapbox dropped the `interactive` property: https://github.com/mapbox/mapbox-gl-js/issues/1479
-export function getInteractiveLayerIds(mapStyle) {
```

Comment:
btw, `interactive` and any other keywords that are not recognized by Mapbox will break their style diffing (it wouldn't not know how to translate those keywords to map actions). We should also remove those properties when we pass the style to Mapbox.